src/artifactminer/RepositoryIntelligence/repo_intelligence_user.py

```
@@ -0,0 +1,145 @@
+#Part of the Repository Intelligence Module
+#Owner: Evan/van-cpu
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Optional, List
+from pathlib import Path
+import git
+from artifactminer.db.models import Consent, UserRepoStat, UserAIntelligenceSummary
+from artifactminer.db.database import SessionLocal
+from src.artifactminer.RepositoryIntelligence.repo_intelligence_main import isGitRepo, Pathish
+from email_validator import validate_email
+from artifactminer.helpers.openai import get_gpt5_nano_response
+
+@dataclass
+class UserRepoStats:
+    project_name: str 
+    first_commit: Optional[datetime] = None 
+    last_commit: Optional[datetime] = None 
+    total_commits: Optional[int] = None 
+    userStatspercentages:  Optional[int] = None# Percentage of user's contributions compared to total repo activity
+    commitFrequency: Optional[float] = None # Average number of commits per week by the user
+
+
+def getUserRepoStats(repo_path: Pathish, user_email: str) -> UserRepoStats: 
+    if not isGitRepo(repo_path): 
+        raise ValueError(f"The path {repo_path} is not a git repository.") 
+    if not validate_email(user_email):
+        raise ValueError(f"The email {user_email} is not valid.")
```

`email_validator.validate_email` raises `EmailNotValidError` instead of returning `False.` This means an invalid address bypasses the intended `ValueError` and bubbles up the library-specific exception, so callers get inconsistent error handling and the message defined here is never reached.

Something like this:

```python
try:
    user_email = validate_email(user_email, check_deliverability=False).email
except EmailNotValidError as e:
    raise ValueError(f"Invalid email address: {user_email}") from e
```
Here we did `check_deliverability=False` as we dont need to look up DNS records for our email.